src/PSM3.sol

```diff
@@ -60,6 +77,26 @@ contract PSM3 is IPSM3 {
         require(_asset1Precision <= 1e18, "PSM3/asset1-precision-too-high");
     }
 
+    /**********************************************************************************************/
+    /*** Owner functions                                                                        ***/
+    /**********************************************************************************************/
+
+    function setPocket(address newPocket) external override onlyOwner {
+        require(newPocket != address(0), "PSM3/invalid-pocket");
+
+        uint256 amountToTransfer = asset0.balanceOf(pocket);
```

Comment:
`pocket` is used multiple times. I would cache it as a local variable to remove SLOADs.

Reply:
Fixed